File: backend/app/schemas/persona.py
"""Persona Template Schemas"""

from uuid import UUID
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, Field, field_validator


class PersonaBase(BaseModel):
    """Base persona template schema"""
    name: str = Field(..., min_length=1, max_length=100)
    role_type: str = Field(..., min_length=1, max_length=50)
    personality_traits: list[str] = Field(default_factory=list)
    communication_style: str = Field(..., min_length=1, max_length=500)
    persona_metadata: dict = Field(default_factory=dict)
    display_order: int = Field(default=0, ge=0)

    @field_validator('personality_traits', mode='before')
    @classmethod
    def ensure_personality_traits_list(cls, v):
        if v is None:
            return []
        return v

    @field_validator('persona_metadata', mode='before')
    @classmethod
    def ensure_persona_metadata_dict(cls, v):
        if v is None:
            return {}
        return v

    @field_validator('display_order', mode='before')
    @classmethod
    def ensure_display_order_int(cls, v):
        if v is None:
            return 0
        return v


class PersonaCreate(PersonaBase):
    """Schema for creating a new persona template"""
    pass


class PersonaUpdate(BaseModel):
    """Schema for updating a persona template"""
    name: Optional[str] = Field(None, min_length=1, max_length=100)
    role_type: Optional[str] = Field(None, min_length=1, max_length=50)
    personality_traits: Optional[list[str]] = None
    communication_style: Optional[str] = Field(None, min_length=1, max_length=500)
    persona_metadata: Optional[dict] = None
    display_order: Optional[int] = Field(None, ge=0)
    is_active: Optional[bool] = None


class PersonaResponse(PersonaBase):
    """Schema for persona template response"""
    id: UUID
    is_active: bool
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True


class PersonaWithUsageStats(PersonaResponse):
    """Persona with usage statistics"""
    active_agents_count: int = 0
    total_agents_created: int = 0

File: frontend/src/types/persona.ts
export interface PersonaTemplate {
  id: string
  name: string
  role_type: string
  avatar?: string | null
  personality_traits: string[]
  communication_style: string
  persona_metadata: Record<string, any>
  is_active: boolean
  display_order: number
  created_at: string
  updated_at: string
}

export interface PersonaCreate {
  name: string
  role_type: string
  avatar?: string | null
  personality_traits: string[]
  communication_style: string
  persona_metadata?: Record<string, any>
  display_order?: number
}

export interface PersonaUpdate {
  name?: string
  role_type?: string
  avatar?: string | null
  personality_traits?: string[]
  communication_style?: string
  persona_metadata?: Record<string, any>
  display_order?: number
  is_active?: boolean
}

export interface PersonaWithUsageStats extends PersonaTemplate {
  active_agents_count: number
  total_agents_created: number
}

export type RoleType =
  | "team_leader"
  | "business_analyst"
  | "developer"
  | "tester"

export const roleTypeLabels: Record<RoleType, string> = {
  team_leader: "Team Leader",
  business_analyst: "Business Analyst",
  developer: "Developer",
  tester: "Tester",
}

export const roleTypeColors: Record<RoleType, string> = {
  team_leader: "bg-purple-500/10 text-purple-400 border-purple-500/20",
  business_analyst: "bg-blue-500/10 text-blue-400 border-blue-500/20",
  developer: "bg-green-500/10 text-green-400 border-green-500/20",
  tester: "bg-amber-500/10 text-amber-400 border-amber-500/20",
}
